### DISCUSSION TITLE
Running interactive flows with "expect" script in Actions

### DISCUSSION THREAD

**Discussion by khitrenovich** [community/community#25044] [Actions]
<p>Hi all,</p>
<p>I have an interactive 3rd party Node.js script to be run as part of a job. I’ve automated the responses with <code>(auto)expect</code> and trying to run <code>expect</code> script instead. That works fine locally, but hangs in Actions - relevant step never completes (and shows no output when running).</p>
<p>After some investigation I’ve found that <code>expect</code>-ing Bash scripts works, it is Node script that causes issues.</p>
<p>I have reproduced the issue in separate clean repo, where you can see “shell” job passes and “node” job hangs - <a href="https://github.com/khitrenovich/github-actions-expect-test/actions" rel="nofollow noopener">https://github.com/khitrenovich/github-actions-expect-test/actions.</a></p>
<p>Any advices are welcome.</p>
<p>Thanks in advance!</p>

**Comment by khitrenovich** [comment, marked as answer]
<p>After digging more into that, I managed to trace the problem down to the special formatting characters that “expect” was… well, expecting. Running the original Node program on my Ubuntu box resulted in prettier prompt than the prompt that was (supposedly) generated on “dumb” terminal in Actions. Changing the expect script to wait for the plain prompt text solved the issue for me.</p>
<p>I realize it’s not “one size fits all” solution, but at least I know that in general “expect+Node.js” combination is usable in GitHub Actions.</p>

**Comment by akefirad** [comment]
Hi @khitrenovich would you kindly share your solution (or the clean repo you created before). Anything that can help me to figure out how to do this would be greatly appreciated. Thanks.

**Reply by khitrenovich** [reply]
We are more than 5 years since then, I'm two jobs older and hardly remember what it was about.

**Reply by akefirad** [reply]
😁Fair enough, I had a little hope you might still have it. Thanks anyway.
